multisite-directory.php

```diff
@@ -213,37 +212,37 @@ private function initializeDirectory () {
     }
 
     /**
-     * Outputs site directory category fields during new site signup on front-end.
+     * Outputs site directory category fields during new site signup on front-end or the Network Admin screen.
      *
      * @link https://developer.wordpress.org/reference/hooks/signup_blogform/
+     *
+     * @uses $pagenow to detect if the form is on the Network Admin add new site screen and output matching HTML
      */
     public static function signup_blogform () {
+        global $pagenow;
         require_once ABSPATH.'wp-admin/includes/template.php';
         $html = wp_terms_checklist(0, array(
             'taxonomy' => Multisite_Directory_Taxonomy::name,
             'echo'     => false,
         ));
-        print '<div id="multisite-directory-signup-categories">';
-        print '<label>'.__('Site Categories:', 'multisite-directory').'</label>';
-        print str_replace("disabled='disabled'", '', $html);
-        print '</div><!-- #multisite-directory-signup-categories -->';
-    }
+        $html = str_replace("disabled='disabled'", '', $html);
+        $label = '<label>'.__('Site Categories:', 'multisite-directory').'</label>';
 
-    /**
-     * Adds the site categories to the default site creation meta variables.
-     *
-     * @link https://developer.wordpress.org/reference/hooks/add_signup_meta/
-     *
-     * @param array $meta
-     */
-/*
-    public static function add_signup_meta ($meta) {
-        if (isset($_POST['tax_input']) && !empty($_POST['tax_input'][Multisite_Directory_Taxonomy::name])) {
-            $meta['multisite-directory-signup-categories'] = $_POST['tax_input'][Multisite_Directory_Taxonomy::name];
+        print '<div id="multisite-directory-signup-categories">';
+        if ($pagenow == 'site-new.php') {
```

Comment:
Just a minor note, if you're not familiar with this convention, consider reading about "[Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions)." The short version is that:

```php
'site-new.php' == $pagenow
```

is "better" than the other way around (the way you have it):

```php
$pagenow == 'site-new.php'
```

The reason it's "better" with the variable to the *right* of the conditional operator is because you protect the code from an accidental deletion of one of the `=` characters, which would change the meaning of the statement from a condition to an assignment (`$pagenow = 'site-new.php'`). With the variable to the right of the conditional, this same mistake would cause a fatal error rather than re-assigning to the `$pagenow` variable, because you cannot assign a variable to a string literal: `'site-new.php = $pagenow // fatal error`.

This kind of defensive programming is especially important when dealing with global variables. [Yoda conditions in PHP are also a WordPress coding standard](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions).

Anyway, I don't think you need to change this per se, I've already marked it on my local branch for modification. I'm just giving you a heads up about it in case you don't already know about Yoda  conditions so you can start using them in the future and improve your own coding practice.